odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -2313,6 +2313,39 @@ def test_51_search_many2one_ordered(self):
         ''']):
             self.env['test_new_api.message'].search([], order='discussion')
 
+    def test_52_search_many2one_active_test(self):
+        Model = self.env['test_new_api.model_active_field']
+        parent_active = Model.create({
+            'name': 'Parent',
+        })
+        parent_inactive = Model.create({
+            'active': False,
+            'name': 'Parent',
+        })
+        child_parent_active = Model.create({
+            'parent_id': parent_active.id,
+        })
+        child_parent_inactive = Model.create({
+            'parent_id': parent_inactive.id,
+        })
+
+        self.assertEqual(
+            Model.search([('parent_id.name', '=', 'Parent')]),
+            child_parent_active + child_parent_inactive,
+        )
+        self.assertEqual(
+            Model.search([('parent_id', '=', 'Parent')]),
+            child_parent_active + child_parent_inactive,
+        )
+        # Weird semantic: parent_active is in both results but he didn't have a parent_id
```

Comment:
Indeed, parent_active should not be returned.

Reply:
We will change that later :+1: